plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -286,6 +292,122 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static <T extends Type> TypeCoercer<DecimalType, T> createDecimalToInteger(DecimalType fromType, T toType)
+    {
+        if (fromType.isShort()) {
+            return new ShortDecimalToIntegerCoercer<>(fromType, toType);
+        }
+        return new LongDecimalToIntegerCoercer<T>(fromType, toType);
+    }
+
+    private static class LongDecimalToIntegerCoercer<T extends Type>
+            extends TypeCoercer<DecimalType, T>
+    {
+        private final long minValue;
+        private final long maxValue;
+
+        public LongDecimalToIntegerCoercer(DecimalType fromType, T toType)
+        {
+            super(fromType, toType);
+
+            if (toType.equals(TINYINT)) {
+                minValue = Byte.MIN_VALUE;
+                maxValue = Byte.MAX_VALUE;
+            }
+            else if (toType.equals(SMALLINT)) {
+                minValue = Short.MIN_VALUE;
+                maxValue = Short.MAX_VALUE;
+            }
+            else if (toType.equals(INTEGER)) {
+                minValue = Integer.MIN_VALUE;
+                maxValue = Integer.MAX_VALUE;
+            }
+            else if (toType.equals(BIGINT)) {
+                minValue = Long.MIN_VALUE;
+                maxValue = Long.MAX_VALUE;
+            }
+            else {
+                throw new TrinoException(NOT_SUPPORTED, format("Could not create Coercer from Decimal to %s", toType));
+            }
```

Comment:
Can we extract them into a abstract class that which could implement `applyCoercedValue` and we could introduce a abstract method which could generate string equivalent to the decimal type.